src/formatDate.js

```diff
@@ -50,7 +50,51 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const result = [];
+
+  const oldSeperator = fromFormat.pop();
+  const dateArray = date.split(oldSeperator);
+  const newSeperator = toFormat.pop();
+
+  let year, month, day;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].includes('Y')) {
+      year = dateArray[i];
+    }
+
+    if (fromFormat[i].includes('M')) {
+      month = dateArray[i];
+    }
+
+    if (fromFormat[i].includes('D')) {
+      day = dateArray[i];
+    }
+  }
+
+  for (let i = 0; i < dateArray.length; i++) {
+    if (toFormat[i].includes('Y')) {
+      if (toFormat[i].length === 4) {
+        if (parseInt(year.slice(-2)) < 30) {
```

Comment:
You can also use + sign what is a bit shorter but less visible, to be honest I dont know what is better option here

Reply:
Its a tiny detail but I'll change it